src/mono/mono/metadata/gc-internals.h

```diff
@@ -430,4 +430,11 @@ extern gboolean mono_do_not_finalize;
 /* List of names of classes not to finalize. */
 extern gchar **mono_do_not_finalize_class_names;
 
+/*
+ * Unified runtime stop/restart world, SGEN Only.
+ * Will take and release the LOCK_GC.
+ */
+void mono_stop_world (MonoThreadInfoFlags flags);
```

Comment:
Why wouldn't mono_gc_stop_world work here? I see that this calls unified_suspend_stop_world, but I am not sure I appreciate what the differences are in the different ways to stop the world.

Reply:
mono_gc_stop_world currently have a number of side effects, mostly stats/events and logging, but also doing some pre-work on togglerefs etc, so its not a pure STW feature and using that from the sample profiler, that does this at very high frequency will cause to much side effects so we needed a cleaner STW, closer to unified_suspend_stop_world. We also need the ability to better control the exclude threads compared to what's done during regular GC STW. I currently made mono_stop_world/mono_restart_world internal to the runtime and in the long run I would like the code to move out from SGEN into a runtime specific feature (STW is not GC specific feature). I guess we could potentially re-wire mono_gc_stop_world into just calling internal mono_stop_world, but if we like mono_gc_stop_world to be close to regular GC STW then I believe we should keep it as is.

Reply:
@lateralusX `mono_gc_stop_world` is very new and has only one caller that has nothing to do with GC.  It was added basically just to add an easy to use STW api.  If you update its only caller to use `mono_stop_world`, you can get rid of it.

Reply:
OK, cool, so we could then remove mono_gc_stop_world/mono_gc_restart_world from mono-gc.h and switch external GC call sites into using mono_stop_world/mono_start_world. Maybe we should put those functions in some other internal runtime header? @lambdageek any suggestions?

Reply:
@lateralusX `threads-types.h`, maybe. Although I don't feel particularly strongly about it one way or another.

Reply:
Replaced mono_gc_stop|restart_world with mono_stop|restart_world. Kept in gc-internals.h for now until we move generic STW code out of GC codebase.